packages/core/lib/package.js

```diff
@@ -1,351 +1,214 @@
       );
     }
 
-    const pkg = new EthPM(options.working_directory, host, registry);
-
-    if (options.packages) {
-      const promises = options.packages.map(package_name => {
-        const pieces = package_name.split("@");
-        package_name = pieces[0];
-
-        let version = "*";
-
-        if (pieces.length > 1) version = pieces[1];
-
-        return pkg.installDependency(package_name, version);
-      });
-
-      Promise.all(promises)
-        .then(() => {
-          if (options.packages.length > 0) {
-            console.log("");
-            console.log("Successfully installed the following package(s)...");
-            console.log("==================================================");
-            options.packages.forEach(singlePackage => {
-              console.log(`> ${singlePackage}`);
-            });
-            console.log("");
-          }
-          callback();
-        })
-        .catch(callback);
-    } else {
-      fs.access(
-        path.join(options.working_directory, "ethpm.json"),
-        fs.constants.R_OK,
-        error => {
-          let manifest;
-
-          // If the ethpm.json file doesn't exist, use the config as the manifest.
-          if (error) manifest = options;
-
-          pkg
-            .install(manifest)
-            .then(() => {
-              callback();
-            })
-            .catch(callback);
-        }
-      );
-    }
+    // Fetch target package manifest URI
+    const manifestURI = await ethpm.registries
+      .package(ethpmURI.packageName)
+      .release(ethpmURI.version);
+
+    // Install target manifest URI to working directory
+    await ethpm.installer.install(manifestURI, ethpmURI.address);
+    options.logger.log(
+      `Installed ${ethpmURI.packageName}@${ethpmURI.version} to ${
+        options.working_directory
+      }`
+    );
   },
 
-  publish: function(options, callback) {
-    var self = this;
-
+  release: async options => {
+    expect.options(options.ethpm, [
+      "ipfs_host",
+      "ipfs_port",
+      "ipfs_protocol",
+      "registry"
+    ]);
     expect.options(options, [
-      "ethpm",
       "working_directory",
       "contracts_directory",
-      "networks"
+      "contracts_build_directory",
+      "networks",
+      "network",
+      "logger"
     ]);
 
-    expect.options(options.ethpm, ["registry", "ipfs_host"]);
-
-    // ipfs_port and ipfs_protocol are optinal.
-
-    // When publishing, you need a ropsten network configured.
-    var ropsten = options.networks.ropsten;
-
-    if (!ropsten) {
-      return callback(
-        new TruffleError(
-          "You need to have a `ropsten` network configured in order to publish to the Ethereum Package Registry. See the following link for an example configuration:" +
-            OS.EOL +
-            OS.EOL +
-            "    http://truffleframework.com/tutorials/using-infura-custom-provider" +
-            OS.EOL
-        )
+    if (options.network !== "mainnet") {
```

Comment:
Note that the user defines custom names for networks, so `"mainnet"` here is jut a magic string